### ISSUE TITLE
stops recording after few hours

### ISSUE THREAD

**Issue by donavan9** [andrey-leshenko/WaTracker#11]
Sir you wrote a nice application. Thanks a ton for your effort. I am testing for few days the issue is it stops recording in 2 to 4 hours. I have to relaunch.  It says stream stopped etc. My phone is active no network issues. Could you help. Thx

**Comment by Idontknowdude** [comment]
Can confirm. There is an issue that WaTracker stops recording (at random, mostly once a day for me) however the software still thinks it is recording properly (this is indicated by the "wa_stream_start" messages). Restarting the recording window solves the problem, tho this isn't a real solution ofc, because you can't monitor the software all the time.

**Comment by andrey-leshenko** [comment]
Hello,

Sorry to hear about the problem.
Currently I don't have time to work on the app,
but if someone figures our the solution I'll be happy to update the code.

-Andrey

**Comment by Dit007** [comment]
Hello, very nice extension but is there already a solution to the problem or a workaround that works automatically?
Last week it ran smoothly and at the weekend stops when wa_stream_end appears in the log file which happens more often. I can not even pursue an hour ... THX to the Devloper for the work and would be glad if the extension works again... Best regards

**Comment by Fueidi2** [comment]
really great extension, too bad it doesnt work for me too.
i used it for 3 weeks without any problem and since a few days it only works for 5-60min and then you get the wa_stream_end message.
i tried it on a different computer, with different phone numbers and even an older version of this extension. nothing fixed it.
i would be really happy if the extension will work again in the near future :)

**Comment by v1v3kgithub** [comment]
I don't think the issue related to stopped recording is being cased by the tool, its seems more like a change on Wassup service that checks to ensure that the user is using the browser. I would assume the idea being if the user hasn't used the web app (say no messages sent, no mouse movements etc it would stop responding)

**Comment by SpaceTraveller1** [comment]
I think simply faking some mouse clicks in the web app window would solve this.

**Comment by erik1o6** [comment]
So using my logitech keyboard I generated a infinite left click script that ran every second and the application still goes into Signed in but not recording

**Comment by erik1o6** [comment]
Well it runs for me for about 30 mins to an hour I simulated clicks within the chrome app loading different chats every 5 seconds but I still get signed in but not recording. So we can at least rule out the user input theory. The only solution is to close the record window and hit the record button again. 
I'm not a JavaScript developer or chrome app dev so I don't really understand where the code execution starts. So could someone update the current script to close the record window and relaunch it every ten minutes or so and we should be able to get around the recording timing out. If someone can tell me which files I need to modify I should be able to do it myself and push the changes for everyone.

I'm also getting the following as a error message
Message dropped because port is closed

**Comment by erik1o6** [comment]
So I wrote a script using sikulix and it basically just closes the recording applications and reopens it every 10 mins and watracker works flawlessly but the solution is definitely not optimal as the recording window has to be always opened and I have to have a computer screen always dedicated for watracker so that I can use the script to track any changes on the recording

**Comment by donavan9** [comment]
I even tried mouse clicker, but itstill does not help. Using windows script i even tried to reload, rerecord but still it stops
Message dropped because port is closed or stream stopped etc

Could you please help?

**Comment by donavan9** [comment]
can you please share the script am not a developer?

**Comment by erik1o6** [comment]
Okay will do

This is the idea

while True:
    while not exists():
        sleep(5)
    doubleClick()
    type("q")
    click()
    wait(5)
I'm at work so I will send the full script after

**Comment by donavan9** [comment]
how will i execute this loop? and how does it start? Where I should embed this code?

**Comment by erik1o6** [comment]
Hi Andrey,

Just install sikulix on your machine open it up with the gui and run it.
The only requirement is that you can only the watracker windows open in
your Main Desktop(if you have manY) and it should work right away. If it
doesnt work just use the gui to edit the images because they might not work
since mine have a different resolution. Let me know if you make any
modification.

Cheers,
Erik Arfvidson

**Comment by erik1o6** [comment]
he only requirement is that you can only the watracker windows open in your
Main Desktop(if you have manY) and it should work right away.

Lol thats not clear the only req is that you have the watracker windows
open in your main display. So basically the watracker launcher and the
record window. The code is pretty straight forward to follow with sikulix
gui

Cheers,
Erik Arfvidson

**Comment by MAISTA23** [comment]
Even tho i installed sikulix and used the code above, its just does nothing. There seems something to be missing. Care to eloberate on this? Thanks ;)

**Comment by erik1o6** [comment]
It works for me. I mean you might need to debug it. Remove the infinite
while loop and see if it closes the record window when it says signed in
but not recording. If I had to guess  my resolution is probably higher than
yours and that why sikulix is not working for you (so just retake the
images).

However, again this is a fix that makes it work but it's not a solution.
Can a JavaScript developer help us find a permanent solution?

if I understood the flow of the program I would do the following for a
solution:
1.I would check when the record windows says "Signed in but not recording"
2. kill the record window
3. relaunch the record window event when you hit the record button in the
main menu

Hope that helps!

Let me know if you have any other additional questions?

Cheers,
Erik

**Comment by MAISTA23** [comment]
Figured it out. Thanks alot. Now WaTracker is not working at all anymore (since like 2 hours). Its just says port is closed or something like that immediately after i wanna start recording. Reinstalled (even the other fork) and restartet everything. Not working ;(

**Comment by erik1o6** [comment]
Yeah mine says that all the time but it still works

**Comment by MAISTA23** [comment]
It is still working for you? I just woke up and tried again. Nothing. Hes not starting the record. Even tried to reinstall chrome. 

![unbenannt](https://user-images.githubusercontent.com/28226054/38665861-c1060e7e-3e3d-11e8-97e5-c29404c49d60.png)

![unbenannt1](https://user-images.githubusercontent.com/28226054/38669128-db9fdeec-3e45-11e8-887b-9b02a45a5b5c.png)


This on repeat. Worked fine yesterday tho

**Comment by erik1o6** [comment]
Yeah it's working for me without issue I'll send you a screen shot when I
get home

**Comment by MAISTA23** [comment]
> Yeah it's working for me without issue I'll send you a screen shot when I
get home

Damn. Whats happening then for me? :(

**Comment by Fueidi2** [comment]
same for me.
worked till yesterday, today the port is closed just from the start :(

**Comment by SpaceTraveller1** [comment]
Yeah I have the same issue

**Comment by erik1o6** [comment]
I can confirm my solution still works

**Comment by MAISTA23** [comment]
> I can confirm my solution still works

So. Everyone else got the same error but you? Im using Windows 10, reinstalled the tracker, chrome and whatsapp itself on my phone. Tried it on another PC. Nothing works. It just wont record anymore.

**Comment by erik1o6** [comment]
I think we got banned or something. Cause mine stopped working just now

**Comment by MAISTA23** [comment]
> I think we got banned or something. Cause mine stopped working just now

Then its not an account ban. I tried it on another whatsapp account. Same error.

**Comment by donavan9** [comment]
me to face the same problem I used to restart the record button multiple times it worked for a day and then no I get Signed in not recording. I reinstalled on multiple machines. getting the same issue. Can some one help?

**Comment by donavan9** [comment]
it looks like something is happening to block this

**Comment by alferyo** [comment]
I had the same problem.
Notice that when "wa_stream_end" occurred in the log it said that it was throwing the messages because the port was closed, but the messages were coming.
So edit the script and modify it. I deleted
case 'wa_stream_end':
if (port) {
port.disconnect ();
port = null;
}
break;
So the port does not close.
I've been without problems for weeks

**Comment by alferyo** [comment]
I have another problem. The activity of some contacts is never recorded, but if I consult through the whstsapp on the phone, I can see the last connection or if there is the line, but wa tracker only recoerd it on a few occasions., 
Can someone help? thanks

**Comment by MAISTA23** [comment]
> I have another problem. The activity of some contacts is never recorded, but if I consult through the whstsapp on the phone, I can see the last connection or if there is the line, but wa tracker only recoerd it on a few occasions.,
Can someone help? thanks

It somehow records only contacts who have an active conversation tab open. There was some workaround here, look for  it.

So it is working for you as we speak? I tried your solution and it does not work

**Comment by donavan9** [comment]
I keep getting the following message and it is not recording but in the logs I see the messages for login and logout timing, But it is not recording it shows  "Signed in but no Recording" 

i almost spend a day but unable to resolve can you please help?

**Comment by donavan9** [comment]
any help on this please now I am getting stream.store.markUnavailable blocked, what is the reason
